### ISSUE TITLE
unable to log kong api logs to syslog via syslog plugin

### ISSUE THREAD

**Issue by iconist** [Mashape/kong#2764]
I use the syslog plugin available with kong api gateway and I have the following:

```
{ "api_id": "some_id",
 "id": "some_id",
 "created_at": 4544444,
 "enabled": true,
 "name": "syslog",
 "config":
           { "client_errors_severity": "info",
             "server_errors_severity": "info", 
             "successful_severity": "info", 
             "log_level": "emerg"
            } }
```

I use a centos7 ,and i have the following conf file(/etc/rsyslog.conf)

```
# rsyslog configuration file

# For more information see /usr/share/doc/rsyslog-*/rsyslog_conf.html
# If you experience problems, see http://www.rsyslog.com/doc/troubleshoot.html

#### MODULES ####

# The imjournal module bellow is now used as a message source instead of imuxsock.
$ModLoad imuxsock # provides support for local system logging (e.g. via logger command)
$ModLoad imjournal # provides access to the systemd journal
#$ModLoad imklog # reads kernel messages (the same are read from journald)
#$ModLoad immark  # provides --MARK-- message capability

# Provides UDP syslog reception
$ModLoad imudp
$UDPServerRun 514

# Provides TCP syslog reception
$ModLoad imtcp
$InputTCPServerRun 514


#### GLOBAL DIRECTIVES ####

# Where to place auxiliary files
$WorkDirectory /var/lib/rsyslog

# Use default timestamp format
$ActionFileDefaultTemplate RSYSLOG_TraditionalFileFormat
$template precise,"%syslogpriority%,%syslogfacility%,%timegenerated%,%HOSTNAME%,%syslogtag%,%msg%\n"
$ActionFileDefaultTemplate precise

# File syncing capability is disabled by default. This feature is usually not required,
# not useful and an extreme performance hit
#$ActionFileEnableSync on

# Include all config files in /etc/rsyslog.d/
$IncludeConfig /etc/rsyslog.d/*.conf

# Turn off message reception via local log socket;
# local messages are retrieved through imjournal now.
$OmitLocalLogging on

# File to store the position in the journal
$IMJournalStateFile imjournal.state


#### RULES ####

# Log all kernel messages to the console.
# Logging much else clutters up the screen.
#kern.*                                                 /dev/console

# Log anything (except mail) of level info or higher.
# Don't log private authentication messages!
*.info;mail.none;authpriv.none;cron.none                /var/log/messages

# The authpriv file has restricted access.
authpriv.*                                              /var/log/secure

# Log all the mail messages in one place.
mail.*                                                  -/var/log/maillog


# Log cron stuff
cron.*                                                  /var/log/cron

# Everybody gets emergency messages
*.emerg                                                 :omusrmsg:*

# Save news errors of level crit and higher in a special file.
uucp,news.crit                                          /var/log/spooler

# Save boot messages also to boot.log
local7.*                                                /var/log/boot.log


# ### begin forwarding rule ###
# The statement between the begin ... end define a SINGLE forwarding
# rule. They belong together, do NOT split them. If you create multiple
# forwarding rules, duplicate the whole block!
# Remote Logging (we use TCP for reliable delivery)
#
# An on-disk queue is created for this action. If the remote host is
# down, messages are spooled to disk and sent when it is up again.
#$ActionQueueFileName fwdRule1 # unique name prefix for spool files
#$ActionQueueMaxDiskSpace 1g   # 1gb space limit (use as much as possible)
#$ActionQueueSaveOnShutdown on # save messages to disk on shutdown
#$ActionQueueType LinkedList   # run asynchronously
#$ActionResumeRetryCount -1    # infinite retries if host is down
# remote host is: name/ip:port, e.g. 192.168.0.1:514, port optional
#*.* @@remote-host:514
# ### end of the forwarding rule ###

```

Syslog daemon is running on the centos 7 machine and it's configured with logging level severity(info) same as or lower than the set config.log_level(emerg).

I am unable to see any logs on
`/var/log/messages`

**Comment by p0pr0ck5** [comment]
This sounds like an issue with your syslog configuration, not Kong. I don't believe this is the appropriate forum to troubleshoot this issue. You may want use something like the `logger` utility to verify your syslog setup: http://help.papertrailapp.com/discussions/questions/96-how-to-log-a-message-from-the-linux-command-line.html

**Comment by iconist** [comment]
Logger Facility is working fine. Able to log a message from the linux command line.

**Comment by satishmane** [comment]
I added /sbin/syslogd in entrypoint script of docker. It solved issue of syslogd daemon not getting started. Now syslog plugin logs messages to /var/log/messages. But these messages are truncated. On further investigation, it seems that kong alpine image uses musl libc which has hardcoded buffer/string size which is not enough. I tried with kong centos version too, but no luck. Instead of syslogd, I installed rsyslogd as part of docker entrypoint script, it improved length of messages getting logged. But still complete message is not getting logged. I tried adding MaxMessageSize setting in rsyslog.conf, but that did not help… any idea how to resolve this? Otherwise syslog plug-in in kong is useless if underlying syslog issue as result of alpine is not resolved
